src/main/java/org/opentripplanner/updater/vehicle_rental/datasources/GbfsVehicleRentalDataSource.java

```diff
@@ -31,214 +34,144 @@
+        brstation.isCarStation = routeAsCar;
+
+        GBFSRentalUris rentalUris = station.getRentalUris();
+        if (rentalUris != null) {
+            String androidUri = rentalUris.getAndroid();
+            String iosUri = rentalUris.getIos();
+            String webUri = rentalUris.getWeb();
+            brstation.rentalUris = new VehicleRentalStationUris(androidUri, iosUri, webUri);
         }
 
-        @Override
-        public VehicleRentalStation makeStation(JsonNode stationNode) {
-            VehicleRentalStation brstation = new VehicleRentalStation();
-            brstation.id = new FeedScopedId(networkName, stationNode.path("station_id").asText());
-            brstation.longitude = stationNode.path("lon").asDouble();
-            brstation.latitude = stationNode.path("lat").asDouble();
-            brstation.name =  new NonLocalizedString(stationNode.path("name").asText());
-            brstation.isKeepingVehicleRentalAtDestinationAllowed = config.allowKeepingRentedVehicleAtDestination();
-            brstation.isCarStation = routeAsCar;
-
-            if (stationNode.has("rental_uris")) {
-                var rentalUrisObject = stationNode.path("rental_uris");
-                String androidUri = rentalUrisObject.has("android") ? rentalUrisObject.get("android").asText() : null;
-                String iosUri = rentalUrisObject.has("ios") ? rentalUrisObject.get("ios").asText() : null;
-                String webUri = rentalUrisObject.has("web") ? rentalUrisObject.get("web").asText() : null;
-                brstation.rentalUris = new VehicleRentalStationUris(androidUri, iosUri, webUri);
-            }
-
-            return brstation;
-        }
+        return brstation;
     }
 
-    class GbfsStationStatusDataSource extends GenericJsonVehicleRentalDataSource<GbfsVehicleRentalDataSourceParameters> {
-
-        public GbfsStationStatusDataSource (GbfsVehicleRentalDataSourceParameters config) {
-            super(config, "data/stations");
+    public VehicleRentalStation mapStationStatus(GBFSStation station, String system) {
+        VehicleRentalStation brstation = new VehicleRentalStation();
+        brstation.id = new FeedScopedId(system, station.getStationId());
+        brstation.vehiclesAvailable = (int) (double) station.getNumBikesAvailable();
+        if (station.getNumDocksAvailable() != null) {
+            brstation.spacesAvailable = (int) (double) station.getNumDocksAvailable();
         }
+        brstation.isKeepingVehicleRentalAtDestinationAllowed = allowKeepingRentedVehicleAtDestination;
+        brstation.isCarStation = routeAsCar;
+        return brstation;
+    }
 
-        @Override
-        public VehicleRentalStation makeStation(JsonNode stationNode) {
+    public VehicleRentalStation mapFreeBike(GBFSBike bike, String system) {
+        if ((bike.getStationId() == null || bike.getStationId().isBlank()) &&
```

Comment:
FYI: This is a place where using the blank line before the code block is optional (even preferred), because the indentation separate the method declaration and the if statement.